### PR TITLE
Update create_thirdparty_resource example

### PR DESCRIPTION
**Pull request by djkonro**
Issue: https://github.com/kubernetes-incubator/client-python/issues/274

### CODE REVIEW

**Comment by djkonro** [comment]
@mbohlool here is the response I am getting

**Comment by djkonro** [comment]
@mbohlool Below is the output I am getting

```json
{'api_version': 'extensions/v1beta1',
 'description': None,
 'kind': 'ThirdPartyResource',
 'metadata': {'annotations': None,
              'cluster_name': None,
              'creation_timestamp': u'2017-08-07T19:56:32Z',
              'deletion_grace_period_seconds': None,
              'deletion_timestamp': None,
              'finalizers': None,
              'generate_name': None,
              'generation': None,
              'labels': None,
              'name': 'git.k8s.com',
              'namespace': None,
              'owner_references': None,
              'resource_version': '1446098',
              'self_link': '/apis/extensions/v1beta1/thirdpartyresourcesgit.k8s.com',
              'uid': '82ce1b0e-7baa-11e7-8ca2-0800277d3a21'},
 'versions': [{'name': 'v1'}]}
```
I noticed the line ` 'self_link': '/apis/extensions/v1beta1/thirdpartyresourcesgit.k8s.com'` and it seems there should be '/' separating `thirdpartyresources` and `git.k8s.com`.

**Comment by mbohlool** [comment]
I think the example should use this class: https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/client/apis/custom_objects_api.py

Having said that, I would like to know what is the API you are trying to call, can you first try to make the example work with custom_objects_api and then see what is the difference between that class and `create_third_party_resource` API call here? My guess is we won't be using the later and we may want to remove the API in favor of custom_objects_api class.

**Comment by sebgoa** [comment]
@mbohlool are CRD's already supported in the client ?

if yes we should have a notebook that shows CRD not TPR.

**Comment by mbohlool** [comment]
Yes, it is supported. Let's create an issue for that. @djkonro are you interested in doing that?

**Comment by djkonro** [comment]
@mbohlool I would work on the notebook for CRD. I have created the issue https://github.com/kubernetes-incubator/client-python/issues/321. I have also updated the above example to use https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/client/apis/custom_objects_api.py as seen in https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/, just like you suggested above.

**Comment by mbohlool** [comment]
@djkonro Thanks for doing both. Did you actually tried the example to make sure it is working?

**Comment by djkonro** [comment]
@mbohlool Yes, I tried the example and it worked fine.

**Comment by djkonro** [comment]
I think it works just for Kubernetes version 1.7.0 or higher.

**Comment by mbohlool** [comment]
well this is backward compatible with TPRs as they have subset of CRD functionality. I believe it should work with 1.6 maybe with a little modification.

**Comment by mbohlool** [comment]
also this only creates TPRs/CRDs. Do you think you can also create/delete an object of that type?

**Comment by djkonro** [comment]
@mbohlool I have updated the code, but I think the `CustomObjectsApi->create_namespaced_custom_object` creates both a resource and an object of its type, the first time it is called with new type and the next time the function is called the objects are just added to the existing type.

**Comment by djkonro** [comment]
@mbohlool Sorry for the confusion. I have now understood what you implied. I would update the patch soon.

**Comment by djkonro** [comment]
@mbohlool I just realized my earlier patch only worked because I did not know I had created a crontabs custom resource from a YAML file some time earlier. After deleting the crontabs resource, it stopped working. I have updated the patch following the existing example https://github.com/kubernetes-incubator/client-python/blob/master/examples/create_thirdparty_resource.md while using https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/client/apis/custom_objects_api.py and I think it now works fine.

File: examples/create_custom_resource.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from __future__ import print_function
```
Comment (sebgoa) [inline examples/create_custom_resource.py]: it needs an ASL v2.0 license header. check `create_deployment` for example

File: examples/create_custom_resource.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+extensions_api_instance = client.ExtensionsV1beta1Api()
+
+tpr = client.V1beta1ThirdPartyResource()
+tpr.api_version = "extensions/v1beta1"
+tpr.kind = "ThirdPartyResource"
+tpr.metadata = client.V1ObjectMeta(name="repo.git.k8s.com")
+tpr.versions = [client.V1beta1APIVersion(name="v1")]
+
+try:
+    # Create custom Resource
```
Comment (sebgoa) [inline examples/create_custom_resource.py]: This is actually "creating a Third party resource"

File: examples/create_custom_resource.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    api_response = custom_api_instance.create_namespaced_custom_object(
+        group, version,
+        namespace, plural,
+        body=body)
+    pprint(api_response)
+except ApiException as e:
+    print("Exception when calling"
+          " CustomObjectsApi->create_namespaced_custom_object: %s\n" % e)
+
+try:
```
Comment (sebgoa) [inline examples/create_custom_resource.py]: so if I run this script, I delete the custom object just after creating it. there is no way I can check manually that it was created in the first place.

**Comment by sebgoa** [comment]
ok this works quite nicely, but there is a creation and a deletion back to back. so the result is this:

```
$ python ./create_custom_resource.py 

Creating custom resource ...
{'api_version': 'extensions/v1beta1',
 'description': None,
 'kind': 'ThirdPartyResource',
 'metadata': {'annotations': None,
              'cluster_name': None,
              'creation_timestamp': datetime.datetime(2017, 8, 27, 12, 20, 13, tzinfo=tzutc()),
              'deletion_grace_period_seconds': None,
              'deletion_timestamp': None,
              'finalizers': None,
              'generate_name': None,
              'generation': None,
              'initializers': None,
              'labels': None,
              'name': 'repo.git.k8s.com',
              'namespace': None,
              'owner_references': None,
              'resource_version': '26429',
              'self_link': '/apis/extensions/v1beta1/thirdpartyresources/repo.git.k8s.com',
              'uid': '13cc76f5-8b22-11e7-b5ff-08002711d9eb'},
 'versions': [{'name': 'v1'}]}

Creating custom object ...
{u'apiVersion': u'git.k8s.com/v1',
 u'branch': u'branch',
 u'kind': u'Repo',
 u'metadata': {u'creationTimestamp': u'2017-08-27T12:20:23Z',
               u'name': u'blog-repo',
               u'namespace': u'default',
               u'resourceVersion': u'26445',
               u'selfLink': u'/apis/git.k8s.com/v1/namespaces/default/repos/blog-repo',
               u'uid': u'19c53269-8b22-11e7-b5ff-08002711d9eb'},
 u'password': u'password',
 u'repo': u'github.com/user/my-blog',
 u'username': u'username'}

Deleting custom object ...
{u'apiVersion': u'v1',
 u'details': {u'group': u'extensions',
              u'kind': u'thirdpartyresourcedatas',
              u'name': u'blog-repo',
              u'uid': u'19c53269-8b22-11e7-b5ff-08002711d9eb'},
 u'kind': u'Status',
 u'metadata': {},
 u'status': u'Success'}
```

The TPR remains behind:

```
$ kubectl get thirdpartyresources repo.git.k8s.com -o yaml
apiVersion: extensions/v1beta1
kind: ThirdPartyResource
metadata:
  creationTimestamp: 2017-08-27T12:20:13Z
  name: repo.git.k8s.com
  resourceVersion: "26429"
  selfLink: /apis/extensions/v1beta1/thirdpartyresources/repo.git.k8s.com
  uid: 13cc76f5-8b22-11e7-b5ff-08002711d9eb
versions:
- name: v1
```

I think I would put the delete in a separate example script and add a deletion of the TPR as well.
so we have `create_custom_resource.py` and `delete_custom_resource.py`

**Comment by sebgoa** [comment]
@djkonro can you address the comments ?

**Comment by djkonro** [comment]
@sebgoa I added the ` delete_custom_resource.py` as suggested https://github.com/kubernetes-incubator/client-python/pull/335

**Comment by mbohlool** [comment]
@djkonro some of the comment is still not addressed here (and some of these comments in applicable in #335 like the license header). Would you please address the remaining comments and also rebase your branch? This looks good after addressing comments, but I would wait for @sebgoa's LGTM before merging it.

**Comment by mbohlool** [comment]
@djkonro friendly ping.

**Comment by djkonro** [comment]
@mbohlool Sorry for the delayed response, I have been travelling and did not have much free time. I would  start working on this and https://github.com/kubernetes-incubator/client-python/pull/335 now.

**Comment by mbohlool** [comment]
@djkonro no problem. Good to have you back.

**Comment by djkonro** [comment]
@mbohlool , I have updated the code

**Pull request closed without merging**